pkg/controller/dns/external_dns_reconciler.go

```diff
@@ -20,19 +25,71 @@ func newExternalDnsReconciler(manager ctrl.Manager, resources []client.Object) e
 
 // NewExternalDns starts all resources required for external dns
 func NewExternalDns(manager ctrl.Manager, conf *config.Config, self *appsv1.Deployment) error {
-	if conf.DNSZoneDomain == "" {
+	if len(conf.DNSZoneIDs) == 0 {
 		return nil
 	}
 
-	dnsConfig := &manifests.ExternalDnsConfig{
-		ResourceName:  "external-dns",
-		TenantId:      conf.TenantID,
-		Subscription:  conf.DNSZoneSub,
-		ResourceGroup: conf.DNSZoneRG,
-		Domain:        conf.DNSZoneDomain,
-		RecordId:      conf.DNSRecordID,
-		IsPrivate:     conf.DNSZonePrivate,
-	}
-	objs := manifests.ExternalDnsResources(conf, self, dnsConfig)
+	configs := parseZoneIds(conf)
+	// one config for private, one config for public
+	var objs []client.Object
+
+	objs = append(objs, manifests.ExternalDnsResources(conf, self, configs)...)
+
 	return newExternalDnsReconciler(manager, objs)
 }
+
+func parseZoneIds(conf *config.Config) (configs []*manifests.ExternalDnsConfig) {
+	var ret []*manifests.ExternalDnsConfig
+	var privateZones, publicZones []string
+	var privateSubscription, privateZoneRg, publicSubscription, publicZoneRg string
+
+	for _, zoneId := range conf.DNSZoneIDs {
+		parsedZone, err := azure.ParseResourceID(zoneId)
+		// this should be impossible
+		if err != nil {
```

Comment:
we should at least log a warning

Reply:
after discussion offline, returning error here